### PR TITLE
Create user

### PR DESCRIPTION
**Pull request by kabcouwer**
## Implemented Features:
- Create user mutation

## What (if anything) did you refactor?
- none

## Roadblocks/Logs/Errors:
- that was tough!

## Testing:
- 100% coverage
- All tests passing

## Is there anything that you need from your teammate?
- nothing at all :)

## Notes/For the future:
- none

## Type of change
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas

### CODE REVIEW

**Review by gunnarrunner — APPROVED** [review]
Good work this will make it easier to get the update a user good to go so this is amazing work@

File: app/graphql/mutations/base_mutation.rb [hunk 1]
```diff
@@ -1,8 +1,5 @@
 module Mutations
-  class BaseMutation < GraphQL::Schema::RelayClassicMutation
-    argument_class Types::BaseArgument
-    field_class Types::BaseField
-    input_object_class Types::BaseInputObject
-    object_class Types::BaseObject
+  class BaseMutation < GraphQL::Schema::Mutation
```
Comment (MTCawthray) [inline app/graphql/mutations/base_mutation.rb:2]: Looks like a good class right there.

File: app/graphql/mutations/users/create_user.rb [hunk 2]
```diff
@@ -1,21 +1,20 @@
...
-      argument :username, String, required: true
       argument :auth_provider, AuthProviderSignupData, required: false
 
       type Types::UserType
 
-      def resolve(username: nil, auth_provider: nil)
+      def resolve(auth_provider: nil)
         User.create!(
-          username: username,
+          username: auth_provider&.[](:credentials)&.[](:username),
```
Comment (MTCawthray) [inline app/graphql/mutations/users/create_user.rb:14]: This looks like syntax that accesses the nested areas to add our arguments. V cool!

File: spec/graphql/mutations/create_new_user_spec.rb [hunk 3]
```diff
@@ -27,8 +27,8 @@ def query
           user: createUser(
           authProvider: {
             credentials: {
-              username: 'Gertie'
-              password: 'password'
+              username: "Gertie",
```
Comment (MTCawthray) [inline spec/graphql/mutations/create_new_user_spec.rb:30]: Are these hard coded right now? If I get a successful query will I see this as the user?

**Review by MTCawthray — APPROVED** [review]
I don't understand much of it at all and I respect what you all do! APPROVED!

**Pull request merged**